std/variant.d

```diff
@@ -2049,8 +2068,8 @@ unittest
     Algebraic!(int, float, string) variant2 = 5.0f;
     // Shouldn' t compile as float not handled by visitor.
     static assert(!__traits(compiles, variant2.visit!(
-                        (int) {},
-                        (string) {})()));
+                        (int i) {},
```

Comment:
This change is necessary due to [an apparent DMD bug](https://github.com/D-Programming-Language/phobos/pull/4035/files) I discovered while working on this.